aten/src/ATen/native/native_functions.yaml

```diff
@@ -686,6 +686,14 @@
 - func: cat.names_out(Tensor[] tensors, Dimname dim, *, Tensor(a!) out) -> Tensor(a!)
   supports_named_tensor: True
 
+- func: block_diag(Tensor[] tensors) -> Tensor
+  supports_named_tensor: True
```

Comment:
In its current state `block_diag` does not look like it supports named tensor. We should delete this line to disable support for named tensors.